service/src/test/java/bio/terra/workspace/app/configuration/external/controller/FolderApiControllerTest.java

```diff
@@ -0,0 +1,310 @@
+    ApiFolderDescription secondFolder =
+        createFolder(workspaceId, displayName, description, firstFolder.getId());
+
+    assertEquals(displayName, secondFolder.getDisplayName());
+    assertEquals(description, secondFolder.getDescription());
+    assertEquals(firstFolder.getId(), secondFolder.getParentFolderId());
+    assertNotNull(secondFolder.getId());
+  }
+
+  @Test
+  public void getFolder_returnsFolderDescription() throws Exception {
+    UUID workspaceId = createDefaultWorkspace(mockMvc, objectMapper).getId();
+    var displayName = "foo";
+    var description = String.format("This is folder %s", displayName);
+
+    ApiFolderDescription firstFolder =
+        createFolder(workspaceId, displayName, description, /*parentFolderId=*/ null);
+
+    String folderGetResponse =
+        mockMvc
+            .perform(
+                addAuth(
+                    get(String.format(FOLDER_V1_PATH_FORMAT, workspaceId, firstFolder.getId())),
+                    USER_REQUEST))
+            .andExpect(status().is(HttpStatus.SC_OK))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    ApiFolderDescription retrievedFolder =
+        objectMapper.readValue(folderGetResponse, ApiFolderDescription.class);
+
+    assertEquals(firstFolder, retrievedFolder);
+  }
+
+  @Test
+  public void getFolder_invalidFolder_throws404() throws Exception {
+    UUID workspaceId = createDefaultWorkspace(mockMvc, objectMapper).getId();
+
+    mockMvc
+        .perform(
+            addAuth(
+                get(String.format(FOLDER_V1_PATH_FORMAT, workspaceId, UUID.randomUUID())),
+                USER_REQUEST))
+        .andExpect(status().is(HttpStatus.SC_NOT_FOUND));
+  }
+
+  @Test
+  public void listFolders_listAllFoldersInAWorkspace() throws Exception {
+    UUID workspaceId = createDefaultWorkspace(mockMvc, objectMapper).getId();
+    var displayName = "foo";
+    var description = String.format("This is folder %s", displayName);
+
+    ApiFolderDescription firstFolder =
+        createFolder(workspaceId, displayName, description, /*parentFolderId=*/ null);
+    ApiFolderDescription secondFolder =
+        createFolder(
+            workspaceId, displayName, description, /*parentFolderId=*/ firstFolder.getId());
+
+    String foldersGetResponse =
+        mockMvc
+            .perform(addAuth(get(String.format(FOLDERS_V1_PATH_FORMAT, workspaceId)), USER_REQUEST))
+            .andExpect(status().is(HttpStatus.SC_OK))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    var retrievedFolders =
+        objectMapper.readValue(foldersGetResponse, ApiFolderDescriptionsList.class);
+
+    var expectedFolders =
+        new ApiFolderDescriptionsList().addFoldersItem(firstFolder).addFoldersItem(secondFolder);
+    assertEquals(expectedFolders, retrievedFolders);
+  }
+
+  @Test
+  public void updateFolders_folderUpdated() throws Exception {
```

Comment:
Can you have

// Assert description is updated
updateFolder_moveToTopLevelFalse_folderNotMoved

// Assert description is updated and folder moved
updateFolder_moveToTopLevelTrue_folderMoved